web/text_layer_builder.js

```diff
@@ -331,6 +334,32 @@ class TextLayerBuilder {
     this.renderMatches(this.matches);
   }
 
+  /**
+   * @private
+   */
+  _bindEvents() {
+    const { eventBus, _boundEvents, } = this;
+
+    _boundEvents.pageCancelled = (evt) => {
+      if (evt.pageNumber !== this.pageNumber) {
+        return;
+      }
+      if (this.textLayerRenderTask) {
+        console.error('TextLayerBuilder._bindEvents: `this.cancel()` should ' +
+                      'have been called when page rendering was cancelled.');
+        return;
+      }
+      for (const name in _boundEvents) {
```

Comment:
Could you add a comment here about why this is necessary? Basically a few lines of summary from the discussion in the other PR so we don't forget the reason why we added this and don't accidentally remove this later without a better alternative.

Reply:
Good point; please check that the added comment is clear enough.

Reply:
Looks really good!